cmd/upgrade/machinepool/cmd.go

```diff
@@ -53,9 +54,14 @@ var Cmd = &cobra.Command{
 	Args: func(_ *cobra.Command, argv []string) error {
```

Comment:
@Mischulee How about you make the source of the `machinepool` name a `bool` on your `NewUpgradeArgsFunction`?  So something like:

```
func NewUpgradeArgsFunction(flagSource bool) func(cmd *cobra.Command, argv []string) error {
	return func(cmd *cobra.Command, argv []string) error {
		var machinepoolID string
		var err error

		if flagSource {
			flags := cmd.Flags()
			if machinepoolID, err = flags.GetString(machinePoolFlagName); err != nil {
				return err
			}
		} else {
			if len(argv) != 1 {
				return InvalidMachinePoolIdentifier
			}
			machinepoolID = argv[0]
		}

		if !machinepool.MachinePoolKeyRE.MatchString(machinepoolID) {
			return InvalidMachinePoolIdentifier
		}
		return nil
	}
}
```

That means we have a single function for all scenarios, and you can add a few more unit tests to verify the flow via the args as well as via the flag.

Reply:
You are right, this looks good and now it will be more reusable. I added more unit tests for this.